lock/lock_test.go

```diff
@@ -0,0 +1,43 @@
+package lock
+
+import (
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/go-commons/logging"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAcquireLockWithRetries(t *testing.T) {
+	t.Parallel()
+
+	var options = Options {
+		AwsRegion: "us-east-1",
+		LockTable: "test-dynamodb-lock-table",
+		LockString: "test-dynamodb-lock-string-" + random.UniqueId(),
+		MaxRetries: 2,
+		SleepBetweenRetries: 1 * time.Second,
+		Logger: logging.GetLogger("TestAcquireLockWithRetries"),
+	}
+
+	defer assertLockReleased(t, &options)
+	defer ReleaseLock(&options)
+
+	options.Logger.Infof("Acquiring first lock...\n")
+	err := AcquireLock(&options)
+	require.NoError(t, err)
+
+	options.Logger.Infof("Acquiring second lock...\n")
+	err = AcquireLock(&options)
+
+	assert.Error(t, err, "Acquiring of second lock succeeded, but it was expected to fail")
```

Comment:
I thought `assert.Error`, as the 3rd argument, allowed you to check the contents of the error? In this case, we'd want to be sure that the error we got was that the lock was already acquired, and not some other error (e.g., failure to talk to DynamoDB).

Reply:
Yeah, that's correct. This is a leftover actually from the first time around. I'll change it to
```suggestion
		assert.Error(t, err,
		fmt.Sprintf("Error acquiring lock %s in table %s in region %s (already locked?): %s\n",
			options.LockString,
			options.LockTable,
			options.AwsRegion,
			err))
```